### PR TITLE
Dev/notsahlo/recursive rendering

### PR DESCRIPTION
**Pull request by NOTSAHLO**

### CODE REVIEW

File: vue-app/src/components/UI/Recursive.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        :depth = "props.depth + 1"
+        :maxDepth = "props.maxDepth"
+        />
+    </div>
+</template>
+
+<script setup lang="ts">
+import type { RecursiveProps } from "../../models/RecursiveProps"
+import { defineProps, defineComponent } from "vue";
+import Recursive from "./Recursive.vue"
```
Comment (RZEPI) [inline vue-app/src/components/UI/Recursive.vue]: this component name is against the convention

File: vue-app/src/components/UI/Recursive.vue
Comment (RZEPI) [inline vue-app/src/components/UI/Recursive.vue:1]: with higher amounth of elements the page is unreadable

File: vue-app/src/components/UI/Recursive.vue [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const props = defineProps<RecursiveProps>();
+defineComponent({
```
Comment (RZEPI) [inline vue-app/src/components/UI/Recursive.vue]: using defineComponent is redundant

File: vue-app/src/router.ts [hunk 3]
```diff
@@ -2,11 +2,13 @@ import { createRouter, createWebHistory } from "vue-router";
 import HomePage from "./pages/HomePage.vue";
 import TablePage from "./pages/TablePage.vue";
 import NotFound from "./pages/NotFound.vue";
+import RecursivePage from "./pages/RecursivePage.vue";
 
 export default createRouter({
   history: createWebHistory(),
   routes: [
     { path: "/", component: HomePage },
+    { path: "/RecursiveRendering", component: RecursivePage },
```
Comment (RZEPI) [inline vue-app/src/router.ts]: page url should be kebap-case

**Review by RZEPI — CHANGES_REQUESTED** [review]

File: vue-app/src/components/UI/Recursive.vue (hunk 1 above)
Comment (NOTSAHLO) [reply vue-app/src/components/UI/Recursive.vue]: leftover from the creation process, change applied

File: vue-app/src/components/UI/Recursive.vue (hunk 2 above)
Comment (NOTSAHLO) [reply vue-app/src/components/UI/Recursive.vue]: removed redundancy

File: vue-app/src/router.ts (hunk 3 above)
Comment (NOTSAHLO) [reply vue-app/src/router.ts]: path changed to Recursive-Rendering

File: vue-app/src/components/UI/Recursive.vue
Comment (NOTSAHLO) [reply vue-app/src/components/UI/Recursive.vue:1]: is setting the same font size for all recursive components a viable fix to that issue?

Comment (RZEPI) [reply vue-app/src/components/UI/Recursive.vue:1]: yes, but it also is kinda boring, I suggested on React branch my idea to solve it

File: vue-app/src/router.ts (hunk 3 above)
Comment (RZEPI) [reply vue-app/src/router.ts]: kebap-case means that every word should be lowercase

File: vue-app/src/router.ts [hunk 4]
```diff
@@ -2,11 +2,13 @@ import { createRouter, createWebHistory } from "vue-router";
 import HomePage from "./pages/HomePage.vue";
 import TablePage from "./pages/TablePage.vue";
 import NotFound from "./pages/NotFound.vue";
+import RecursivePage from "./pages/RecursivePage.vue";
 
 export default createRouter({
   history: createWebHistory(),
   routes: [
     { path: "/", component: HomePage },
+    { path: "/Recursive-Rendering", component: RecursivePage },
```
Comment (RZEPI) [inline vue-app/src/router.ts]: kebap-case!!!

File: vue-app/src/router.ts (hunk 3 above)
Comment (NOTSAHLO) [reply vue-app/src/router.ts]: changed to lowercase

Comment (NOTSAHLO) [reply vue-app/src/router.ts]: Table should be fixed too

**Review by RZEPI — DISMISSED** [review]

File: vue-app/src/components/UI/RecursiveComponent.vue [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    </div>
+</template>
+
+<script setup lang="ts">
+import type { RecursiveProps } from "../../models/RecursiveProps"
+import RecursiveComponent from "./RecursiveComponent.vue"
+const props = defineProps<RecursiveProps>();
+
+const getColor = (depth : number): string => {
+    const hue = 146 + (depth * 3);
```
Comment (marziel2001) [inline vue-app/src/components/UI/RecursiveComponent.vue]: what happens when it is 300?

File: vue-app/src/components/UI/RecursiveComponent.vue [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    const lightness = 50;
+
+    return `hsl(${hue}, ${saturation}%, ${lightness}%)`;
+  };
+</script>
+
+<style scoped>
+.recursive-component {
+  text-align: center;
+  font-size: 30px;
```
Comment (marziel2001) [inline vue-app/src/components/UI/RecursiveComponent.vue]: why there is 30px, in react 13px?. It should be in ems/rems

Comment (NOTSAHLO) [reply vue-app/src/components/UI/RecursiveComponent.vue]: changed to rems

File: vue-app/src/components/UI/RecursiveComponent.vue (hunk 5 above)
Comment (NOTSAHLO) [reply vue-app/src/components/UI/RecursiveComponent.vue]: added 255 safeguard

**Review by RZEPI — CHANGES_REQUESTED** [review]
in case of renering 100k elements the following error occurs. The max cap or solution for that error needed

image

File: vue-app/src/static/sites.ts [hunk 7]
```diff
@@ -19,7 +19,11 @@ export const sites: Array<Site> = [
...
   {
     label: "Table",
-    link: "/Table",
+    link: "/table",
     description: "Table page, page responsible for testing rendering of big amounts of data. It also displays filtering modal.",
   },
+  {
+    label: "RecRndr",
+    link: "/recursive-rendering",
+  },
```
Comment (RZEPI) [inline vue-app/src/static/sites.ts]: No description in the sites, an app wont start

**Review by RZEPI — CHANGES_REQUESTED** [review]

File: vue-app/src/static/sites.ts (hunk 7 above)
Comment (NOTSAHLO) [reply vue-app/src/static/sites.ts]: Added description

**Review by RZEPI — DISMISSED** [review]

**Review by grczk — DISMISSED** [review]

**Review by RZEPI — DISMISSED** [review]

**Pull request merged**